### ISSUE TITLE
[RDF]: No lockout for rhc

### ISSUE THREAD

**Issue by Cheesedip** [Stormforge-gg/Bugtracker-Fordring#1696]
**Description**: 
When you queue for a random heroic with the dungeon finder you will not get saved to that heroic. This means that if for example the daily heroic is Halls of Lightning. If you want to queue for a 'Random' heroic, you will be getting Halls of Lightning the whole day.

In addition, the main reason for not getting a lockout is so that RDF can server more as a catch-up mechanic. But since we are on the first patch it does not really make any sense to have this. As there is nothing to catch up to.

**Current behaviour**:
No lockout for rhc

**Expected behaviour**: 
Please, lockout for rhc

**Comment by Cold-Martijn** [comment]
Question: if you did random heroic Halls of Lightning, can you queue specific Halls of Lightning afterwards? Or only enter it by doing the random heroic again?

**Comment by Cheesedip** [comment]
Yes you can queue specific HoL after getting it in rhc. I did not check walk-in. Could check if you want me to.

**Comment by Aithne99** [comment]
This is supposedly Blizzlike behavior, and making it give lockout is way too much work instead of just... removing the option to queue for random heroics (while still keeping specific queues for heroics).

**Comment by Cold-Martijn** [comment]
Nah, no need to check that.
This is indeed a bug that should be addressed. RHC should lock you out for the dungeon you've done, but it should be able to get the same dungeon when queueing random (skipping the check for a save).

**Comment by Aithne99** [comment]
RHC locking you or not locking you is pretty much inconsequential compared to it bypassing locked dungeons.

**Comment by Aithne99** [comment]
And even retail info is inconsistent on that one
https://www.mmo-champion.com/threads/1208640-How-do-Heroic-lockouts-work

**Comment by Cheesedip** [comment]
> This is supposedly Blizzlike behavior, and making it give lockout is way too much work instead of just... removing the option to queue for random heroics (while still keeping specific queues for heroics).


> RHC locking you or not locking you is pretty much inconsequential compared to it bypassing locked dungeons.

I'm sorry I dont think I fully understand what you mean here. Could you maybe try to explain it futher?

**Comment by Cheesedip** [comment]
> And even retail info is inconsistent on that one https://www.mmo-champion.com/threads/1208640-How-do-Heroic-lockouts-work

I see what you mean. It does look like all the people saying that you don't get saved from rhc in this thread are playing a later expansion though. 

In addition I have found this forum thread where people seem to agree that in wotlk you did have a lockout for rdf.
http://forum.warmane.com/showthread.php?t=397834

**Comment by Cold-Martijn** [comment]
> > And even retail info is inconsistent on that one https://www.mmo-champion.com/threads/1208640-How-do-Heroic-lockouts-work
> 
> I see what you mean. It does look like all the people saying that you don't get saved from rhc in this thread are playing a later expansion though.
> 
> In addition I have found this forum thread where people seem to agree that in wotlk you did have a lockout for rdf. http://forum.warmane.com/showthread.php?t=397834

Threads from other private servers won't count as evidence ;) anyway, I remember this very precisely from retail WotLK, and this supports it:
https://www.engadget.com/2009-12-08-patch-3-3-the-dungeon-finder-guide-page-2.html
"With random mode on, it will allow you to bypass your normal heroic lockouts."
It doesn't exactly say about the specific dungeons being locked... but I am fairly certain that i remember that correct from WotLK times.

**Comment by Cheesedip** [comment]
I'm starting to get a real vibe that they may have changed this into the icc patch, removing the lockout and that before there was a lockout, seeing how different the opinions are online and how many people seem very sure of two very different things. 

I mentioned this earlier on the discord, that I played until ToC in retail and I can't even remember rdf at all so I'm in no position to argue haha.

Also... not complaining but one of my other tickets just got closed because 'it works like this on warmane so its not bugged'. This seems a bit incosistent. But I understand the difficulty in finding a proper source.

**Comment by Aithne99** [comment]
RDF was added in the ICC patch. Right now, it shouldn't even be active, which is why the easiest solution to this absolute nonissue would be to disable it.

**Comment by Cheesedip** [comment]
Personally I'd love not having RDF, but I don't know how the player base will feel about it with every server having it and now that we have had it for such a long time. Without it it will be much harder to gear up new characters, giving people less incentive perhaps to create new character and join this server. So not sure if this is the best decision here with regard to player base.
